### PR TITLE
ci: Fix test coverage with codecov

### PR DESCRIPTION
**Pull request by dplewis**
### New Pull Request Checklist

- [ ] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-iOS-OSX/security/policy).
- [ ] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-iOS-OSX/issues?q=is%3Aissue).

### Issue Description

Code coverage has been broken for a while with a number of factors contributing to it. `actions/codecov@3.1.0` introduced xcode support. I'm not sure if `CODECOV_TOKEN` is in the in the repository secrets

https://github.com/codecov/codecov-action/releases/tag/v3.1.0

### Approach

* Remove GCC gcov flags
* Enable Clang Coverage (lcov)
* Remove deprecated codecov bash uploader
* Remove codecov.yml in favor of default

### TODO
- [ ] Update Readme Badge
- [x] Upload `CODECOV_TOKEN`

### CODE REVIEW

**Comment by dplewis** [comment]
@mtrezza I think it works, I don't have full access to codecov.

https://app.codecov.io/github/parse-community/Parse-SDK-iOS-OSX/commit/f9615fd91c75718a09539f359b85fb138f6e2840/tree

**Comment by mtrezza** [comment]
> I'm not sure if CODECOV_TOKEN is in the in the repository secrets

We don't have a token in the secrets, IIRC that's not necessary.

**Comment by dplewis** [comment]
@mtrezza This is ready for review. Here is what the coverage looks like with everything included

https://app.codecov.io/github/parse-community/Parse-SDK-iOS-OSX/commit/58c9c5be43bb403982effadee84be31d620d18c1/tree

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -68,23 +68,31 @@ jobs:
...
           name: ${{ env.ARTIFACT_NAME }}
           path: |
-            ~/Library/Logs/DiagnosticReports
             ~/Library/Developer/Xcode/DerivedData/Parse-*/Logs/Test
-      - name: Send Codecov
-        run: bash <(curl https://codecov.io/bash)
+      - name: Upload Coverage
+        uses: codecov/codecov-action@v3.1.1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (mtrezza) [inline .github/workflows/ci.yml:93]: This is interesting, we don't have such a secret manually defined in the repo or the org. How come it's set here?

Comment (dplewis) [reply .github/workflows/ci.yml:93]: It’s not needed for public repos, I’ll remove jt

**Comment by dplewis** [comment]
@mtrezza I made the changes. Can you review it again?

**Review by mtrezza — APPROVED** [review]
Looks good! Thanks for this!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [2.3.0](https://github.com/parse-community/Parse-SDK-iOS-OSX/releases/tag/2.3.0)

**Pull request merged**